File: desktop/BeautyEstiva.Desktop/BeautyEstiva.Desktop/Models/DashboardModels.cs
using System.Text.Json.Serialization;

namespace BeautyEstiva.Desktop.Models;

public class DashboardSummary
{
    [JsonPropertyName("todayAppointmentsCount")]
    public int TodayAppointmentsCount { get; set; }

    [JsonPropertyName("upcomingAppointments")]
    public int UpcomingAppointments { get; set; }

    [JsonPropertyName("thisWeekRevenue")]
    public decimal ThisWeekRevenue { get; set; }

    [JsonPropertyName("thisMonthRevenue")]
    public decimal ThisMonthRevenue { get; set; }

    [JsonPropertyName("thisMonthExpense")]
    public decimal ThisMonthExpense { get; set; }

    [JsonPropertyName("totalCustomers")]
    public int TotalCustomers { get; set; }

    [JsonPropertyName("activePackages")]
    public int ActivePackages { get; set; }

    [JsonPropertyName("monthlyTrend")]
    public List<MonthlyTrend> MonthlyTrend { get; set; } = [];

    [JsonPropertyName("topServices")]
    public List<RevenueByGroup> TopServices { get; set; } = [];

    [JsonPropertyName("topStaff")]
    public List<RevenueByGroup> TopStaff { get; set; } = [];

    [JsonPropertyName("customerGrowth")]
    public List<CustomerGrowth> CustomerGrowth { get; set; } = [];

    [JsonPropertyName("statusDistribution")]
    public AppointmentStatusDistribution? StatusDistribution { get; set; }

    [JsonPropertyName("todaySchedule")]
    public List<TodayAppointment> TodaySchedule { get; set; } = [];
}

public class MonthlyTrend
{
    [JsonPropertyName("month")]
    public string Month { get; set; } = string.Empty;

    [JsonPropertyName("revenue")]
    public decimal Revenue { get; set; }

    [JsonPropertyName("expense")]
    public decimal Expense { get; set; }
}

public class RevenueByGroup
{
    [JsonPropertyName("label")]
    public string Label { get; set; } = string.Empty;

    [JsonPropertyName("count")]
    public int Count { get; set; }

    [JsonPropertyName("amountInTry")]
    public decimal AmountInTry { get; set; }
}

public class CustomerGrowth
{
    [JsonPropertyName("month")]
    public string Month { get; set; } = string.Empty;

    [JsonPropertyName("newCustomers")]
    public int NewCustomers { get; set; }

    [JsonPropertyName("totalCustomers")]
    public int TotalCustomers { get; set; }
}

public class AppointmentStatusDistribution
{
    [JsonPropertyName("scheduled")]
    public int Scheduled { get; set; }

    [JsonPropertyName("confirmed")]
    public int Confirmed { get; set; }

    [JsonPropertyName("completed")]
    public int Completed { get; set; }

    [JsonPropertyName("cancelled")]
    public int Cancelled { get; set; }

    [JsonPropertyName("noShow")]
    public int NoShow { get; set; }

    [JsonPropertyName("total")]
    public int Total { get; set; }
}

public class TodayAppointment
{
    [JsonPropertyName("id")]
    public int Id { get; set; }

    [JsonPropertyName("time")]
    public string Time { get; set; } = string.Empty;

    [JsonPropertyName("customerName")]
    public string CustomerName { get; set; } = string.Empty;

    [JsonPropertyName("treatmentName")]
    public string TreatmentName { get; set; } = string.Empty;

    [JsonPropertyName("staffName")]
    public string StaffName { get; set; } = string.Empty;

    [JsonPropertyName("status")]
    public string Status { get; set; } = string.Empty;

    [JsonPropertyName("treatmentColor")]
    public string? TreatmentColor { get; set; }
}

public class BranchListItem
{
    [JsonPropertyName("id")]
    public int Id { get; set; }

    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("address")]
    public string? Address { get; set; }

    [JsonPropertyName("phone")]
    public string? Phone { get; set; }

    [JsonPropertyName("email")]
    public string? Email { get; set; }

    [JsonPropertyName("isMainBranch")]
    public bool IsMainBranch { get; set; }

    [JsonPropertyName("isActive")]
    public bool IsActive { get; set; }

    [JsonPropertyName("staffCount")]
    public int StaffCount { get; set; }
}

File: flutter/lib/core/models/dashboard_models.dart
class DashboardSummary {
  final int todayAppointmentsCount;
  final int upcomingAppointments;
  final double thisWeekRevenue;
  final double thisMonthRevenue;
  final double thisMonthExpense;
  final int totalCustomers;
  final int activePackages;
  final List<MonthlyTrend> monthlyTrend;
  final List<RevenueByGroup> topServices;
  final List<RevenueByGroup> topStaff;
  final List<CustomerGrowth> customerGrowth;
  final StatusDistribution? statusDistribution;
  final List<TodayAppointment> todaySchedule;

  double get netProfit => thisMonthRevenue - thisMonthExpense;

  DashboardSummary({
    required this.todayAppointmentsCount,
    required this.upcomingAppointments,
    required this.thisWeekRevenue,
    required this.thisMonthRevenue,
    required this.thisMonthExpense,
    required this.totalCustomers,
    required this.activePackages,
    required this.monthlyTrend,
    required this.topServices,
    required this.topStaff,
    required this.customerGrowth,
    this.statusDistribution,
    required this.todaySchedule,
  });

  factory DashboardSummary.fromJson(Map<String, dynamic> json) {
    return DashboardSummary(
      todayAppointmentsCount: json['todayAppointmentsCount'] ?? 0,
      upcomingAppointments: json['upcomingAppointments'] ?? 0,
      thisWeekRevenue: (json['thisWeekRevenue'] ?? 0).toDouble(),
      thisMonthRevenue: (json['thisMonthRevenue'] ?? 0).toDouble(),
      thisMonthExpense: (json['thisMonthExpense'] ?? 0).toDouble(),
      totalCustomers: json['totalCustomers'] ?? 0,
      activePackages: json['activePackages'] ?? 0,
      monthlyTrend: (json['monthlyTrend'] as List? ?? [])
          .map((e) => MonthlyTrend.fromJson(e))
          .toList(),
      topServices: (json['topServices'] as List? ?? [])
          .map((e) => RevenueByGroup.fromJson(e))
          .toList(),
      topStaff: (json['topStaff'] as List? ?? [])
          .map((e) => RevenueByGroup.fromJson(e))
          .toList(),
      customerGrowth: (json['customerGrowth'] as List? ?? [])
          .map((e) => CustomerGrowth.fromJson(e))
          .toList(),
      statusDistribution: json['statusDistribution'] != null
          ? StatusDistribution.fromJson(json['statusDistribution'])
          : null,
      todaySchedule: (json['todaySchedule'] as List? ?? [])
          .map((e) => TodayAppointment.fromJson(e))
          .toList(),
    );
  }
}

class MonthlyTrend {
  final String month;
  final double revenue;
  final double expense;

  MonthlyTrend({required this.month, required this.revenue, required this.expense});

  factory MonthlyTrend.fromJson(Map<String, dynamic> json) => MonthlyTrend(
        month: json['month'] ?? '',
        revenue: (json['revenue'] ?? 0).toDouble(),
        expense: (json['expense'] ?? 0).toDouble(),
      );
}

class RevenueByGroup {
  final String label;
  final int count;
  final double amountInTry;

  RevenueByGroup({required this.label, required this.count, required this.amountInTry});

  factory RevenueByGroup.fromJson(Map<String, dynamic> json) => RevenueByGroup(
        label: json['label'] ?? '',
        count: json['count'] ?? 0,
        amountInTry: (json['amountInTry'] ?? 0).toDouble(),
      );
}

class CustomerGrowth {
  final String month;
  final int newCustomers;
  final int totalCustomers;

  CustomerGrowth({required this.month, required this.newCustomers, required this.totalCustomers});

  factory CustomerGrowth.fromJson(Map<String, dynamic> json) => CustomerGrowth(
        month: json['month'] ?? '',
        newCustomers: json['newCustomers'] ?? 0,
        totalCustomers: json['totalCustomers'] ?? 0,
      );
}

class StatusDistribution {
  final int scheduled;
  final int confirmed;
  final int completed;
  final int cancelled;
  final int noShow;
  final int total;

  StatusDistribution({
    required this.scheduled,
    required this.confirmed,
    required this.completed,
    required this.cancelled,
    required this.noShow,
    required this.total,
  });

  factory StatusDistribution.fromJson(Map<String, dynamic> json) => StatusDistribution(
        scheduled: json['scheduled'] ?? 0,
        confirmed: json['confirmed'] ?? 0,
        completed: json['completed'] ?? 0,
        cancelled: json['cancelled'] ?? 0,
        noShow: json['noShow'] ?? 0,
        total: json['total'] ?? 0,
      );
}

class TodayAppointment {
  final int id;
  final String time;
  final String customerName;
  final String treatmentName;
  final String staffName;
  final String status;
  final String? treatmentColor;

  TodayAppointment({
    required this.id,
    required this.time,
    required this.customerName,
    required this.treatmentName,
    required this.staffName,
    required this.status,
    this.treatmentColor,
  });

  factory TodayAppointment.fromJson(Map<String, dynamic> json) => TodayAppointment(
        id: json['id'] ?? 0,
        time: json['time'] ?? '',
        customerName: json['customerName'] ?? '',
        treatmentName: json['treatmentName'] ?? '',
        staffName: json['staffName'] ?? '',
        status: json['status'] ?? '',
        treatmentColor: json['treatmentColor'],
      );
}
